core/src/com/biglybt/core/history/impl/DownloadHistoryManagerImpl.java

```diff
@@ -770,29 +770,24 @@ public void perform(TimerEvent event) {
 
 		// System.out.println( "loading " + file );
 
-		try{
-			if ( FileUtil.resilientConfigFileExists( file )){
-
-				Map map = FileUtil.readResilientConfigFile( file );
-
-				List<Map<String,Object>>	list = (List<Map<String,Object>>)map.get( "records" );
+		if (FileUtil.resilientConfigFileExists(file)) {
+			Map map = FileUtil.readResilientConfigFile(file);
 
-				for ( Map<String,Object> m: list ){
+			List<Map<String, Object>> list = (List<Map<String, Object>>) map.get("records");
+			if (list != null) {
 
-					try{
-						DownloadHistoryImpl record = new DownloadHistoryImpl( result, m );
+				for (Map<String, Object> m : list) {
 
-						result.put( record.getUID(), record );
+					try {
+						DownloadHistoryImpl record = new DownloadHistoryImpl(result, m);
+						result.put(record.getUID(), record);
 
-					}catch( Throwable e ){
+					} catch (Throwable e) {
```

Comment:
Since the FileUtil methods are fail safe then this try-catch block should now be sufficiant enough...